src/history/test/SerializeTests.cpp

```diff
@@ -12,23 +12,40 @@ using namespace stellar;
 
 TEST_CASE("Serialization round trip", "[history]")
 {
-    std::vector<std::string> testFiles = {
-        "stellar-history.testnet.6714239.json",
-        "stellar-history.livenet.15686975.json"};
-    for (auto const& fn : testFiles)
+    std::vector<std::string> inputFiles = {
+        "stellar-history.testnet.6714239.input.json",
+        "stellar-history.livenet.15686975.input.json",
+        "stellar-history.testnet.6714239.networkPassphrase.json"};
+    std::vector<std::string> outputFiles = {
+        "stellar-history.testnet.6714239.output.json",
+        "stellar-history.livenet.15686975.output.json",
+        "stellar-history.testnet.6714239.networkPassphrase.json"};
+    for (int i = 0; i < inputFiles.size(); i++)
     {
         std::string fnPath = "testdata/";
-        fnPath += fn;
-        SECTION("Serialize " + fnPath)
+        std::string inputPath = fnPath + inputFiles[i];
+        std::string outputPath = fnPath + outputFiles[i];
+        SECTION("Serialize " + inputPath)
         {
-            std::ifstream in(fnPath);
+            std::ifstream in(inputPath);
             in.exceptions(std::ios::badbit);
-            std::string fromFile((std::istreambuf_iterator<char>(in)),
-                                 std::istreambuf_iterator<char>());
+            std::string input((std::istreambuf_iterator<char>(in)),
+                              std::istreambuf_iterator<char>());
+
+            std::ifstream out(outputPath);
+            out.exceptions(std::ios::badbit);
+            std::string output((std::istreambuf_iterator<char>(out)),
+                               std::istreambuf_iterator<char>());
 
             HistoryArchiveState has;
-            has.fromString(fromFile);
-            REQUIRE(fromFile == has.toString());
+
+            // Test fromString
+            has.fromString(input);
+            REQUIRE(output == has.toString());
+
+            // Test load
+            has.load(inputPath);
```

Comment:
do not use the same `has` for reading and writing in case something goes wrong (like load no-ops)

Reply:
Makes sense! I created a new object `hasLoad` to test `load`.